File: java/FizzBuzz/src/Rule.java
class Rule {
	private final int divisor;
	private final String result;

	Rule(int divisor, String result) {
		this.divisor = divisor;
		this.result = result;
	}

	static RuleSelector divisibleBy(int divisor) {
		return new RuleSelector(divisor);
	}

	boolean appliesTo(int number) {
		return number % divisor == 0;
	}

	String giveResult() {
		return result;
	}
}

File: php/FizzBuzz/src/Rule.php
<?php

namespace Core;


class Rule
{
    private $denominator;
    private $result;

    /**
     * Rule constructor.
     */
    public function __construct($denominator, $result)
    {
        $this->denominator = $denominator;
        $this->result = $result;
    }

    /**
     * @param $number
     * @param $denominator
     * @return bool
     */
    private static function isDivisibleBy($number, $denominator):bool
    {
        return $number % $denominator == 0;
    }

    public function appliesTo($number)
    {
        return self::isDivisibleBy($number, $this->denominator);
    }

    public function giveResult()
    {
        return $this->result;
    }
}